src/utils/string.ts

```
@@ -3,6 +3,216 @@ const ansiRegex = [
 
+/**
+ * Checks if a Unicode code point is a full-width character (occupies 2 columns in a terminal).
+ * Includes CJK ideographs, Hangul syllables, full-width forms, and wide punctuation.
+ *
+ * @param {number} code - The Unicode code point.
+ * @returns {boolean} True if the character is full-width.
+ */
+function isFullWidth(code: number): boolean {
+  return (
+    // CJK Unified Ideographs
+    (code >= 0x4e00 && code <= 0x9fff) ||
```

ESLint

[error] Invalid group length in numeric value.

(unicorn/numeric-separators-style)